### ISSUE TITLE
(cli): CDK CLI is not discovering SSO Credentials 

### ISSUE THREAD

**Issue by sam-goodwin** [aws/aws-cdk#23520]
### Describe the bug

Something has changed from under me that has broken the AWS CDK CLI with SSO. This used to work before.

When I run deploy:
```
cdk deploy --profile=sam
```

I get an error:
```
Unable to determine the default AWS account: ProcessCredentialsProviderFailure: Profile sam did not include credential process
```

I dug into this - it's trying to load ProcessCredentials because the SsoCredentials is failing to load.

If you look at this code in aws-sdk, you'll notice it expects sso_start_url to be on the profile: 

https://github.com/aws/aws-sdk-js/blob/ce9d028875f0766829caa3127436d8ef2fd4abdf/lib/credentials/sso_credentials.js#L105

But after running `aws configure sso`, my `~/aws/config` doesn't contain that information on the profile - it's moved to the sso-session. I think this is new behavior?

```
[profile sam]
sso_session = sam
sso_account_id = <redacted>
sso_role_name = AdministratorAccess
region = us-west-2
output = json
[sso-session sam]
sso_start_url = <redacted>
sso_region = us-east-1
sso_registration_scopes = sso:account:access
```

Moving the properties from sso-session to the profile gets past this error, but then I get another error:

```
SsoCredentialsProviderFailure: Cached credentials not found under sam profile. Please make sure you log in with aws sso login first
```

But I have logged in. After some hacking in the .js of the aws-sdk, I discovered that it's looking for the following cache file:

```
.aws/sso/cache/9616dde3b3a6e3e3cf79ed3a2d8a81d96feb7baa.json
```

But `aws sso login --profile=sam` produces the following files:
```
dee28091556274a08d5d440787aa4df6c6267b17.json
f16bed56189e249fe4ca8ed10a1ecae60e8ceac0.json
```

### Expected Behavior

`cdk deploy --profile=sam` should successfully deploy

### Current Behavior

cdk deploy fails. I have provided details in the description.

### Reproduction Steps

Not sure how to do that. 

### CDK CLI Version

2.50.0 (build 4c11af6)

### Node.js Version

v16.14.2

### OS

Mac OS 11.5.1

### Language

Typescript

**Comment by pahud** [comment]
Hi @sam-goodwin 

I believe you should configure your ~/.aws/config with profile only as below:

```
[profile sam]
sso_account_id = <redacted>
sso_role_name = AdministratorAccess
region = us-east-1
output = json
sso_start_url = <redacted>
sso_region = us-east-1
```

or with both `profile` and `sso-session` sections as below:

```
[profile sam]
sso_session = sam-sso-session
sso_account_id = <redacted>
sso_role_name = AdministratorAccess
region = us-east-1
output = json

[sso-session sam-sso-session]
sso_start_url = <redacted>
sso_region = us-east-1
sso_registration_scopes = sso:account:access
```

Either way, you should be able to run `cdk deploy --profile sam`.

Please let me know if it works with you.

**Comment by sam-goodwin** [comment]
The latter is what was generated when I ran `aws configure sso` but it was still failing. CDK's config doesn't work with the default SSO behavior of my version of the AWS CLI:

```
aws-cli/2.9.0 Python/3.11.0 Darwin/20.6.0 source/arm64 prompt/off
```

**Comment by pahud** [comment]
Hi @sam-goodwin 

If your ~/.aws/config is as below:

```
[profile sam]
sso_session = sam-sso-session
sso_account_id = <redacted>
sso_role_name = AdministratorAccess
region = us-east-1
output = json

[sso-session sam-sso-session]
sso_start_url = <redacted>
sso_region = us-east-1
sso_registration_scopes = sso:account:access
```

1. try execute `aws --profile sam sts get-caller-identity` and you should get a response of your current identity. If it fails, you have to run `aws configure sso --profile sam` again to authenticate and authorize with the named sso session(`sam-sso-session` in my case).
2. if the above step successfully returns, you should be able to run `npx cdk diff --profile sam` on an empty cdk app as below and you should be able to run cdk synth or deploy as well.
<img width="550" alt="image" src="https://user-images.githubusercontent.com/278432/210289060-fc853bf7-27f2-4524-81cf-fed633b528e5.png">


My AWS CLI version is 2.9.12 FYR

```
$ aws --version
aws-cli/2.9.12 Python/3.9.11 Linux/5.15.0-47-generic exe/x86_64.debian.10 prompt/off
```

**Comment by setu4993** [comment]
I was running into the exact same issue with AWS CDK version `2.50.0` and AWS CLI `2.9.0`. I stumbled upon here while trying to figure out how to fix it.

Updating just AWS CLI to `2.9.12` didn't fix the issue.

Then, I updated AWS CDK to `2.59.0` (latest) and that fixed the issue.

I looked through the release notes briefly but I don't see anything about SSO in there, so I'm not sure what was broken / fixed between `2.50.0` and `2.59.0`.
